io/las/LasReader.cpp

```diff
@@ -619,7 +619,17 @@ point_count_t LasReader::readFileBlock(std::vector<char>& buf,
     point_count_t blockpoints = buf.size() / ptLen;
 
     blockpoints = std::min(maxpoints, blockpoints);
+    if (m_istream->eof())
+        throw pdal::invalid_stream("stream is done");
```

Comment:
Not sure I like throwing an exception here.  Also, I've been trying to get rid of all the subclassed exceptions that we're not using anyway (of course I haven't TOLD anyone that I've been doing this.)